scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -386,6 +386,23 @@ def check(method):
     for method in methods:
         check(method)
 
+@pytest.mark.parametrize("method", ['FW', 'J', 'BF'])
+def test_masked_invalid_input(method):
+    # Reference - https://github.com/scipy/scipy/pull/22286
+    csgraph = np.array(
+        [[0, 1, 0],
+        [1, 0, 0],
+        [0, 0, 0]]
+    )
+    csgraph_masked = np.ma.masked_invalid(csgraph)
+
+    zeros = np.zeros((3, 3))
+
+    SP = shortest_path(csgraph_masked, method=method, directed=True,
+                        overwrite=False)
+    assert_array_equal(SP, zeros)
```

Comment:
Sorry one final nit, I think new tests should use `assert_equal` instead as per the numpy docs